dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java

```diff
@@ -414,4 +437,93 @@ public Config getConfig() {
+            GitHub gitHub = GitHubBuilder.fromEnvironment().withConnector(okHttp3Connector).build();
+
+            GHRepository repository = gitHub.getRepository("dockstore/dockstore-cli");
+            ghRelease = repository.getLatestRelease();
+            allReleases = repository.listReleases();
+        } catch (IOException e) {
+            LOG.info("Could not get CLI releases information from GitHub. ", e);
+            throw new CustomWebApplicationException("Could not get CLI  releases information info from GitHub",
+                    HttpStatus.SC_SERVICE_UNAVAILABLE);
+        }
+        String cliLatestVersion = ghRelease.getName();
+        cliInfo.setCliLatestDockstoreScriptDownloadUrl(DOCKSTORE_CLI_RELEASES_URL + "/download/" + cliLatestVersion + "/dockstore");
+        cliInfo.setCliLatestVersion(cliLatestVersion);
+        // Get the latest unstable version
+        // If there is no unstable version return a null as the latest unstable version
+        Optional<String> cliLatestUnstableVersion = getLatestUnstableVersion(allReleases, cliLatestVersion);
+        if (cliLatestUnstableVersion.isPresent()) {
+            cliInfo.setCliLatestUnstableDockstoreScriptDownloadUrl(DOCKSTORE_CLI_RELEASES_URL + "/download/" + cliLatestUnstableVersion.get() + "/dockstore");
+            cliInfo.setCliLatestUnstableVersion(cliLatestUnstableVersion.get());
+        } else {
+            cliInfo.setCliLatestUnstableDockstoreScriptDownloadUrl(null);
+            cliInfo.setCliLatestUnstableVersion(null);
+        }
+        return cliInfo;
+    }
+
+    /**
+     * This method will return the prerelease version that is greater than the latest version if there is one
+     * If there isn't a prerelease version greater than the latest release then don't return one
+     * since we don't want to downgrade to an older untested version
+     *
+     * @param allReleases a PagedIterable of GHRelease objects for releases
+     * @param latestRelease the semantic variable string of the latest release
+     * @return An Optional String of the latest unstable version
+     */
+    private static Optional<String> getLatestUnstableVersion(PagedIterable<GHRelease> allReleases, String latestRelease) {
+        // https://stackoverflow.com/questions/24511052/how-to-convert-an-iterator-to-a-stream
+        Stream<GHRelease> targetStream = StreamSupport.stream(
+                Spliterators.spliteratorUnknownSize(allReleases.iterator(), Spliterator.ORDERED), false);
+        try {
+            return Optional.of(targetStream
+                    .filter(GHRelease::isPrerelease)
+                    .filter(g -> preReleaseVersionIsGreaterThanLatest(g.getName(), latestRelease))
+                    .findFirst().orElseThrow().getName());
+            // orElseThrow throws a NoSuchElementException if there is no prerelease
+        } catch (NullPointerException | NoSuchElementException e) {
```

Comment:
What throws a NullPointerException? It's generally not good to catch those and should be handled in code explicitly.

I think, untested, you could simplify this to:

```
return targetStream
                 .filter(GHRelease::isPrerelease)
                 .findFirst()
                 .filter(g -> preReleaseVersionIsGreaterThanLatest(g.getName(), latestRelease))
                 .map(release -> release.getName();
```
And you don't need the try/catch (except maybe for my question about the NPE).

Reply:
findFirst could generate a NPE if the selected element is null; but I don't think it ever could be null, so removed try catch

Reply:
`findFirst` won't throw an NPE, it will return an `Optional`. If you call `get()` on it and it's empty then you would, but the method itself won't.